api/scpca_portal/config/logging.py

```diff
@@ -50,3 +54,46 @@ def get_and_configure_logger(name: str) -> logging.Logger:
     logger.logger.addHandler(handler)
 
     return logger
+
+
+def get_formatted_time(timestamp: float) -> str:
+    """Return string representation of a timestamp in Hour:Minute:Second:Millisecond format."""
+    return datetime.fromtimestamp(timestamp).strftime("%H:%M:%S.%f")[:-3]  # remove microseconds
+
+
+def log_runtime(logger):
+    """Log the start time, end time, and duration of the wrapped function."""
+
+    def decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            global LOG_RUNTIMES
+            if not LOG_RUNTIMES:
+                return func(*args, **kwargs)
+
+            func_name = f"{func.__module__}::{func.__name__}"
+            start_time = time.time()
+            logger.info(f"\nEntering function '{func_name}'.")
+
+            try:
+                result = func(*args, **kwargs)
+            finally:
+                end_time = time.time()
+                duration = end_time - start_time
+                logger.info(
+                    f"\nExited function '{func_name}'.\n"
+                    f"Function runtime: {get_formatted_time(duration)} "
+                    f"(start time: {get_formatted_time(start_time)}, "
+                    f"end time: {get_formatted_time(end_time)})"
```

Comment:
I have a small suggestion. We could omit `time` from the log message to simplify the output, as it already indicates the function runtime. 

```suggestion
                    f"(start: {get_formatted_time(start_time)}, "
                    f"end: {get_formatted_time(end_time)})"
```

e.g. ) Output
```
Function runtime: 00:00:00.003 (start: 20:33:41.271, end: 20:33:41.274)
```

Reply:
Thanks Nozomi!